File: dartmazing_network/ios/Classes/RequestError.swift
//
//  RequestError.swift
//  dartmazing_network
//
//  Created by Hélio Mesquita on 25/03/21.
//

import Foundation

enum RequestError: Int, Equatable {

    typealias RawValue = Int

    case unknownError = 0
    case invalidParser = 1
    case badRequest = 400
    case unauthorized = 401
    case forbidden = 403
    case notFound = 404

    var message: String {
        return String(describing: self)
    }
    
    var flutterError: FlutterError {
        return FlutterError(code: "\(self.rawValue)", message: message, details: nil)
    }

}

File: dartmazing_network/android/src/main/kotlin/com/dartmazing_network/RequestError.kt
package com.dartmazing_network

enum class RequestError constructor(val statusCode: Int) {
    unknownError(0),
    invalidParser(1),
    badRequest(400),
    unauthorized(401),
    forbidden(403),
    notFound(404);

    companion object {
        fun from(findValue: Int): RequestError = RequestError.values().first { it.statusCode == findValue } ?: RequestError.unknownError
    }
}